### ISSUE TITLE
[@angular-eslint/template/no-inline-styles] conflicts with Angular style guide for [style] object bindings

### ISSUE THREAD

**Issue by MariaMikhnevich** [angular-eslint/angular-eslint#3016]
Body
### Description
The `@angular-eslint/template/no-inline-styles` rule currently reports `[style]` bindings even when the value is an object / style map.
At the same time, Angular's official style guide recommends preferring native `class` and `style` bindings over `ngClass` and `ngStyle`, including object/map binding syntax for `[style]`.
This creates an awkward situation:
- `ngStyle` can be allowed with `allowNgStyle: true`
- `[style.prop]` can be allowed with `allowBindToStyle: true`
- but `[style]="styleObject"` is still reported by the rule
That means teams following the Angular style guide still get lint errors for the recommended native `[style]` syntax.
### Angular Style Guide
Angular recommends:
> Prefer `class` and `style` bindings over using the `NgClass` and `NgStyle` directives.
Example from the style guide:
```html
<div [style]="{'color': textColor, 'background-color': backgroundColor}"></div>
```
https://angular.dev/style-guide#prefer-class-and-style-over-ngclass-and-ngstyle
#### Current behavior
With this config:
```
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error",
      {
        "allowBindToStyle": true
      }
    ]
  }
}
```
This is allowed:
```html
<div [style.color]="textColor"></div>
```
But this is still reported:
```html
<div [style]="{ 'color': textColor, 'background-color': backgroundColor }"></div>
```
#### Expected behavior
It would be useful to support one of the following:
1. Extend allowBindToStyle so it also allows [style] map/object bindings
2. Add a separate option specifically for [style] map/object bindings

For example, something like:
```
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error",
      {
        "allowBindToStyle": true,
        "allowStyleMapBinding": true
      }
    ]
  }
}
```
#### Why this would help
Using [style] with an object is:
- native Angular syntax
- recommended by the Angular style guide over ngStyle
- often more ergonomic than expanding everything into multiple [style.prop] bindings
Right now the main workaround is switching back to ngStyle, but that goes against Angular's own recommendation.

#### Reproduction
```html
<div [style]="{ 'color': textColor, 'background-color': backgroundColor }"></div>
```
```
{
  "rules": {
    "@angular-eslint/template/no-inline-styles": [
      "error",
      {
        "allowBindToStyle": true
      }
    ]
  }
}
```
#### Environment
- `@angular-eslint/eslint-plugin-template`: `21.3.1`

**Comment by xonaib** [comment]
I'd like to work on this. My plan: add a new `allowStyleObjectBinding` option (defaulting to `false` to maintain current behaviour) and split the internal `isNodeHasStyleAttribute` helper into two functions — one for plain `style="..."` attributes (always disallowed) and one for `[style]="obj"` input bindings (optionally allowed via the new option). This mirrors the existing `allowNgStyle` pattern. Does this approach work, or would you prefer extending `allowBindToStyle` to also cover map bindings?

**Comment by MariaMikhnevich** [comment]
@xonaib  
This approach seems reasonable to me. Different projects may have different conventions — some may want to allow `[style]` with an object binding, while others may still want to forbid it even when `allowBindToStyle` is enabled for `[style.prop]`. Having a separate, more granular option (e.g. `allowStyleObjectBinding`) lets teams configure this precisely for their own use case, rather than coupling it to `allowBindToStyle` and forcing an all-or-nothing choice. So I'm all for adding a separate option to allow for more granular customization of the rule.

**Comment by xonaib** [comment]
@MariaMikhnevich, thanks for confirming the preferred approach. I opened PR #3112: https://github.com/angular-eslint/angular-eslint/pull/3112

It adds allowStyleObjectBinding as a separate option, with tests covering [style], [style.prop], and [attr.style] behavior.
